### ISSUE TITLE
[feat] - save last state in off state, so toggling after restart picks up last settings

### ISSUE THREAD

**Issue by jacobw** [swingerman/ha-dual-smart-thermostat#245]
If I restart HA while the climate control is on, it correctly continues as it was before the restart. However, if it's off, it forgets the recent state and I have to manually set it again, instead if just being able to toggle it back on again. This also means I lose the previous temp setting.

Is it possible to save and restore state when of too?

E.g, before restart, last setting was heating with target 20.5. After restart if I toggle on, it goes back to cooling target 19, and I have to go and configure again. This happens fairly regularly due to many updates requiring restarts.

**Comment by swingerman** [comment]
COuld you confirm the current version you use and your current config?

**Comment by jacobw** [comment]
v0.9.7 (though this has been an issue for a while)

I've been trying different things to resolve this issue so a few bits commented out while I've been testing.

```
  - platform: dual_smart_thermostat
    name: Main Bedroom Climate
    unique_id: main_bedroom_climate
    heater: switch.main_bedroom_heatpump_heat
    cooler: switch.main_bedroom_heatpump_cool
    target_sensor: sensor.yz_e871_temperature
    # heat_cool_mode: true
    min_temp: 15
    max_temp: 26
    target_temp: 19.5
    # target_temp_high: 21
    # target_temp_low: 19
    cold_tolerance: 0.5
    hot_tolerance: 0
    precision: 0.1
    target_temp_step: 0.5
```

**Comment by swingerman** [comment]
Thanks. I will look into this.

**Comment by cyberkryten** [comment]
I'm not sure if it is related (or helpful) but hvac_action_reason doesn't get set to anything at all after a restart until the thermostat takes some 'action' again.

So if it was target_temp_reached and you restart (or reload the YAML from Developer tools) then it remains as an empty attribute until something gets changed and causes an action to occur that would change it.

I've been trying to use it to determine if an AC has been turned off by the integration or manually from a control panel and the 'blank' value has caused some issues with the logic

**Comment by jacobw** [comment]
Yeah it's never set every time I check it too. Right now it's active and idle because it just hit target but doesn't say that. Eg

hvac_modes:
  - heat
  - "off"
  - cool
  - heat_cool
min_temp: 15
max_temp: 26
target_temp_step: 0.5
current_temperature: 21.8
temperature: 21
hvac_action: idle
prev_target_temp_low: 21
prev_target_temp_high: 21
prev_target_temp: 21
hvac_action_reason: ""
friendly_name: M Climate
supported_features: 385

**Comment by mat1990dj** [comment]
I have this same issue, for whatever reason even if the UI displays for incance 66, when I turn it on it always goes to 75, which was an old target I had, but I no longer have

https://github.com/user-attachments/assets/8a6ba775-7371-4b06-9129-e22b126e8a2f

All have the same config:

  - platform: dual_smart_thermostat
    name: Kitchen
    unique_id: kitchen
    heater: input_boolean.heat_request_kitchen
    cooler: input_boolean.cool_request_kitchen
    target_sensor: sensor.kitchen_temperature
    humidity_sensor: sensor.kitchen_humidity
    heat_cool_mode: false
    initial_hvac_mode: "off"
    min_temp: 60
    max_temp: 85
    precision: 1
    cold_tolerance: 0.1
    hot_tolerance: 0.1
    min_cycle_duration:
      seconds: 30
    keep_alive:
      minutes: 10
    eco:
        temperature: 68
    home:
        temperature: 66
    comfort:
        temperature: 70
    away:
        temperature: 60
    anti_freeze:
        temperature: 50

**Comment by swingerman** [comment]
> I have this same issue, for whatever reason even if the UI displays for incance 66, when I turn it on it always goes to 75, which was an old target I had, but I no longer have
> 
>  2025-02-21_12-06-22.mp4 
> All have the same config:
> 
> * platform: dual_smart_thermostat
>   name: Kitchen
>   unique_id: kitchen
>   heater: input_boolean.heat_request_kitchen
>   cooler: input_boolean.cool_request_kitchen
>   target_sensor: sensor.kitchen_temperature
>   humidity_sensor: sensor.kitchen_humidity
>   heat_cool_mode: false
>   initial_hvac_mode: "off"
>   min_temp: 60
>   max_temp: 85
>   precision: 1
>   cold_tolerance: 0.1
>   hot_tolerance: 0.1
>   min_cycle_duration:
>   seconds: 30
>   keep_alive:
>   minutes: 10
>   eco:
>   temperature: 68
>   home:
>   temperature: 66
>   comfort:
>   temperature: 70
>   away:
>   temperature: 60
>   anti_freeze:
>   temperature: 50
> 
> Edit: from the dev tools, prev_target_temp is always at 75

I cannot reproduce this with [v0.9.11-beta6](https://github.com/swingerman/ha-dual-smart-thermostat/releases/tag/v0.9.11-beta6)

**Comment by jacobw** [comment]
I haven't noticed this issue lately. Perhaps fixed anciently or something else causing it. Happy to close.

**Comment by swingerman** [comment]
@jacobw Please check [v0.9.12-beta1](https://github.com/swingerman/ha-dual-smart-thermostat/releases/tag/v0.9.12-beta1)
